lib/CodeGen/CGDecl.cpp

```diff
@@ -405,10 +405,10 @@ void CodeGenFunction::EmitStaticVarDecl(const VarDecl &D,
   // a no-op and should not be emitted.
   bool isCudaSharedVar = getLangOpts().CUDA && getLangOpts().CUDAIsDevice &&
                          D.hasAttr<CUDASharedAttr>();
-  bool isHCCTileStaticVar = getLangOpts().CPlusPlusAMP && getLangOpts().DevicePath &&
-                            D.hasAttr<HCCTileStaticAttr>();
+  bool isHCCAcceleratorPath =
+      getLangOpts().CPlusPlusAMP && getLangOpts().DevicePath;
   // If this value has an initializer, emit it.
-  if (D.getInit() && !isCudaSharedVar && !isHCCTileStaticVar)
```

Comment:
This will prevent dynamic initializers from being emitted, this is correct. I do not believe that we actually support this correctly at the moment except for corner cases (and it is actually unclear what the meaning of it is in practice), and I do believe that we would benefit from correctly conveying that (as well as re-assessing the current mechanism for namespace scope variables in [[hc]]). Do we have any case which relies on dynamic initialisation of namespace scope variables referenced from [[hc]] and executes correctly at the moment (I could not identify them). I can re-enable it, but I am uncomfortable about its flakiness and it can create surprises / bugs, IMHO.